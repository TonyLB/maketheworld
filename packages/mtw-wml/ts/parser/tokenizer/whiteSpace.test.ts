import SourceStream from './sourceStream'
import whiteSpaceTokenizer from './whiteSpace'

describe('whiteSpaceTokenizer', () => {
    beforeEach(() => {
        jest.clearAllMocks()
        jest.resetAllMocks()
    })

    it('should return undefined when no match', () => {
        const testStream = new SourceStream('Testing')
        expect(whiteSpaceTokenizer(testStream)).toBe(undefined)
    })
    it('should tokenize whitespace between tags', () => {
        const testStream = new SourceStream(' \n    <Room key=(Test)>')
        expect(whiteSpaceTokenizer(testStream)).toMatchSnapshot()
    })
    it('should tokenize whitespace to end of source', () => {
        const testStream = new SourceStream('    \n\t\t')
        expect(whiteSpaceTokenizer(testStream)).toMatchSnapshot()
        expect(testStream.isEndOfSource).toBe(true)
    })
})