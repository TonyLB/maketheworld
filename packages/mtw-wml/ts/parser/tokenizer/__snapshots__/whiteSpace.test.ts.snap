// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`whiteSpaceTokenizer should tokenize whitespace between tags 1`] = `
Object {
  "endIdx": 5,
  "source": " 
    ",
  "startIdx": 0,
  "type": "Whitespace",
}
`;

exports[`whiteSpaceTokenizer should tokenize whitespace to end of source 1`] = `
Object {
  "endIdx": 6,
  "source": "    
		",
  "startIdx": 0,
  "type": "Whitespace",
}
`;
